cli/pkg/app/nitric.go

```diff
@@ -230,27 +232,54 @@ func (c *NitricApp) Build() error {
 	platformRepository := platforms.NewPlatformRepository(c.apiClient)
 
 	// TODO:prompt for platform selection if multiple targets are specified
-	targetPlatform := appSpec.Targets[0]
+	if len(appSpec.Targets) == 0 {
+		nitricEdit := style.Teal("nitric edit")
+		fmt.Printf("No targets specified in nitric.yaml, run %s to add a target\n", nitricEdit)
+		return nil
+	}
+
+	options := []huh.Option[string]{}
+	for _, target := range appSpec.Targets {
+		options = append(options, huh.NewOption(target, target))
+	}
+
+	var targetPlatform string
+
+	// use a tui selector
+	err = huh.NewSelect[string]().
+		Title("Select a build target").
+		Options(
+			options...,
+		).
+		Value(&targetPlatform).
+		Run()
+	if errors.Is(err, huh.ErrUserAborted) {
+		return nil
+	}
+
+	if targetPlatform == "" {
+		return fmt.Errorf("no target platform selected")
+	}
 
 	platform, err := terraform.PlatformFromId(fs, targetPlatform, platformRepository)
-	if err != nil {
+	if errors.Is(err, terraform.ErrUnauthenticated) {
+		fmt.Println("Please login first, using the `login` command")
+		return nil
+	} else if err != nil {
 		return err
 	}
 
+	fmt.Fprintf(os.Stdout, "\nBuilding application for %s\n\n", targetPlatform)
```

Comment:
Note for future, I think we should probably attach an output writer to the `NitricApp` struct and refactor all the commands to write exclusively to that writer instead of fmt.Print/f/ln. Then in automated tests we can assert the outputs if we like.

Reply:
This is how the terraform engine is currently managing this using `FPrintf`